pkg/cloudprovider/suite_test.go

```diff
@@ -319,21 +344,73 @@ var _ = Describe("CloudProvider", func() {
 			})
 			// Instance is a reference to what we return in the GetInstances call
 			instance.ImageId = aws.String(fake.ImageID())
-			isDrifted, err := cloudProvider.IsMachineDrifted(ctx, machineutil.NewFromNode(node))
+			isDrifted, err := cloudProvider.IsMachineDrifted(ctx, machine)
 			Expect(err).ToNot(HaveOccurred())
 			Expect(isDrifted).To(BeTrue())
 		})
-		It("should not return drifted if the AMI is valid", func() {
-			node := coretest.Node(coretest.NodeOptions{
-				ProviderID: fake.ProviderID(lo.FromPtr(instance.InstanceId)),
+		It("should return drifted if the subnet is not valid", func() {
+			machine := coretest.Machine(v1alpha5.Machine{
+				Status: v1alpha5.MachineStatus{
+					ProviderID: fake.ProviderID(lo.FromPtr(instance.InstanceId)),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1alpha5.ProvisionerNameLabelKey: provisioner.Name,
+						v1.LabelInstanceTypeStable:       selectedInstanceType.Name,
+					},
+				},
+			})
+			instance.SubnetId = aws.String(fake.SubnetID())
+			isDrifted, err := cloudProvider.IsMachineDrifted(ctx, machine)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(isDrifted).To(BeTrue())
+		})
+		It("should not return drifted if the subnet is in the AWSNodeTemplate subnets", func() {
+			machine := coretest.Machine(v1alpha5.Machine{
+				Status: v1alpha5.MachineStatus{
+					ProviderID: fake.ProviderID(lo.FromPtr(instance.InstanceId)),
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						v1alpha5.ProvisionerNameLabelKey: provisioner.Name,
+						v1.LabelInstanceTypeStable:       selectedInstanceType.Name,
+					},
+				},
+			})
+			isDrifted, err := cloudProvider.IsMachineDrifted(ctx, machine)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(isDrifted).To(BeFalse())
+		})
+		It("should return an error if AWSNodeTemplate subnets are empty", func() {
+			machine := coretest.Machine(v1alpha5.Machine{
+				Status: v1alpha5.MachineStatus{
+					ProviderID: fake.ProviderID(lo.FromPtr(instance.InstanceId)),
+				},
 				ObjectMeta: metav1.ObjectMeta{
 					Labels: map[string]string{
 						v1alpha5.ProvisionerNameLabelKey: provisioner.Name,
 						v1.LabelInstanceTypeStable:       selectedInstanceType.Name,
 					},
 				},
 			})
-			isDrifted, err := cloudProvider.IsMachineDrifted(ctx, machineutil.NewFromNode(node))
+			nodeTemplate.Status.Subnets = []v1alpha1.Subnet{}
+			ExpectApplied(ctx, env.Client, nodeTemplate)
+			_, err := cloudProvider.IsMachineDrifted(ctx, machine)
+			Expect(err).To(HaveOccurred())
+		})
+		It("should not return drifted if the machine is valid", func() {
```

Comment:
I'm confused how this one is different than the one that tests whether the subnet is in the set of AWSNodeTemplate subnets

Reply:
Ah you're right, it's the exact same. Let me remove.